ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/AttestationUtilTest.java

```diff
@@ -71,12 +77,56 @@ void setUp(SpecContext specContext) {
         .thenReturn(dataStructureUtil.randomBytes(10));
     when(asyncBLSSignatureVerifier.verify(anyList(), any(Bytes.class), any(BLSSignature.class)))
         .thenReturn(SafeFuture.completedFuture(true));
-    attestationUtil =
-        new AttestationUtil(specVersion.getSchemaDefinitions(), beaconStateAccessors, miscHelpers);
+    attestationUtil = spec.getGenesisSpec().getAttestationUtil();
+
+    switch (specContext.getSpecMilestone()) {
+      case PHASE0:
```

Comment:
Maybe make some light builder and use it here and in `SpecLogic`'s to determinate which util to init?

Reply:
well... in `SpecLogic` we are currently explicitly picking the specific versions, and I'm not for adding production code that  will be used only for test. We may add it together with the other classes if we want to clean up `SpecLogic` creation..

Reply:
I mean something like `AttestationUtil.create(final Milestone milestone)`

Reply:
it should be applied to others too (like `BeaconStateAccessorsAltair`). This PR is already too big, let's avoid it for the moment.